aws/src/main/java/com/xmatters/testing/aws/AWSConnector.java

```diff
@@ -0,0 +1,92 @@
+package com.xmatters.testing.aws;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.DescribeInstancesRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeInstancesResponse;
+import software.amazon.awssdk.services.ec2.model.Instance;
+import software.amazon.awssdk.services.ec2.model.RebootInstancesRequest;
+import software.amazon.awssdk.services.ec2.model.Reservation;
+import software.amazon.awssdk.services.ec2.model.StopInstancesRequest;
+import software.amazon.awssdk.services.ec2.model.Tag;
+import software.amazon.awssdk.services.ec2.model.TerminateInstancesRequest;
+
+public class AWSConnector {
+
+    private static final int INSTANCE_RUNNING = 16;
+
+    private Ec2Client ec2;
+
+    public AWSConnector() {
+        initializeEc2Client();
+    }
+
+    synchronized private void initializeEc2Client() {
+        if (ec2 == null) {
+            ec2 = Ec2Client.create();
+        }
+    }
+
```

Comment:
Still light on comments. On this method it would be nice to mention that only "RUNNING" instances are considered.

```suggestion

    /*
     * Find running instances matching the masks.
     */
```